Source/Authorization/ConsentManager.swift

```diff
@@ -80,23 +102,112 @@ class ConsentManager: NSObject {
         UIViewController.topMostViewController()?.present(viewController, animated: true, completion: nil)
     }
     
-    private func finish(with result: Result<ConsentResponse, Error>) {
+    private func finishUserConsent(with result: Result<ConsentResponse, Error>) {
         guard Thread.current.isMainThread else {
             DispatchQueue.main.async {
-                self.finish(with: result)
+                self.finishUserConsent(with: result)
             }
             return
         }
         
+        reset()
+        
+        userConsentCompletion?(result)
+        userConsentCompletion = nil
+        addServiceCompletion = nil
+    }
+    
+    private func finishAddService(with result: Result<Void, Error>) {
+        guard Thread.current.isMainThread else {
+            DispatchQueue.main.async {
+                self.finishAddService(with: result)
+            }
+            return
+        }
+        
+        reset()
+        
+        addServiceCompletion?(result)
+        addServiceCompletion = nil
+        userConsentCompletion = nil
+    }
+    
+    private func userCancelled() {
+        guard Thread.current.isMainThread else {
+            DispatchQueue.main.async {
+                self.userCancelled()
+            }
+            return
+        }
+        
+        reset()
+        
+        addServiceCompletion?(.failure(ConsentError.userCancelled))
```

Comment:
Duplicated below.

Reply:
No it isn't. They are different callbacks.

Reply:
Should have gone to specsavers 😂